src/BaseNode.ts

```diff
@@ -1,8 +1,14 @@
 export default class BaseNode {
-    private children: BaseNode[];
+    public static PATH_SIZE = 4;
 
-    constructor(private str: string, private data: any = null) {
-        this.children = [];
+    private children: BaseNode[] | null = null;
+
+    constructor(private str: string, private leaf: boolean = false, private data: any = null) {
+    }
+
+    // TODO: change the method name
+    static matchClass(substr: string) {
```

Comment:
What's the purpose of this static method?

Reply:
It is not used yet, in the first iteration we thought to use it for tree construction